### ISSUE TITLE
Can't purchase domain on test API

### ISSUE THREAD

**Issue by oquidave** [gellu/godaddy-api-client#2]
Hey when I try to purchase a domain via the testing url endpoint, I get an error; "missing bill_to_email". Has anyone experienced this?
 
Array[DEBUG] HTTP Response body ~BEGIN~
{"code":"-1073456591","message":"[P3OWGDCOMM002] {gdBasketPurchase.PurchaseBasketXML.1}BasketPurchase has invalid/missing bill_to_email ","name":"ApiError"}
~END~

**Comment by jagroop** [comment]
getting same error :disappointed:

**Comment by ymslavov** [comment]
Same here.

**Comment by anderson-arlen** [comment]
Just so you guys know, this is an api error on godaddy's side of things.

**Comment by jagroop** [comment]
yes , @anderson-arlen  is right, i tried to talk with support team, and they were unable to resolve it ,
they wasted two weeks , i had to scrap my project, !

**Comment by ymslavov** [comment]
That's super weird, isn't this one of their CORE functionalities? How are they overlooking this and decline to put any resources to deal with the issue?

**Comment by oquidave** [comment]
I actually moved on from Godaddy after an attempt at contacting support and getting nothing.

**Comment by anderson-arlen** [comment]
Turns out the API keys generated on developer.godaddy.com are no good for buying domains. If you generate a key on reseller.godaddy.com, it works. Or you can use the same test key that the live documentation uses: `sso-key UzQxLikm_46KxDFnbjN7cQjmw6wocia:46L26ydpkwMaKZV6uVdDWe`

**Comment by ymslavov** [comment]
Hm, I can't get the available() method to work with this key-secret pair
you provided. Is this just for the purchase() method?

**Comment by anderson-arlen** [comment]
sorry, that's the complete auth header. They key pair is:
client id: UzQxLikm_46KxDFnbjN7cQjmw6wocia
secret: 46L26ydpkwMaKZV6uVdDWe

**Comment by ymslavov** [comment]
Yeah, I got that, I just get 401 Unauthorized when I use those client id
and secret. Did you actually get it to work with those?

**Comment by sohaibfarooqi** [comment]
Getting same Error. Turns out you need to set payment method to [Good As Gold](https://sg.godaddy.com/help/what-is-good-as-gold-3359) for purchase to work!
